### PR TITLE
Serializing/Desrializing numpy arrays in Lambda layer arguments

### PR DESCRIPTION
**Pull request by jeremyfix**
When we use numpy arrays in arguments of Lambda layers, they cannot be serialized with model.save.

For example, running 

```
from keras.layers import Input, Lambda
from keras.models import Model
import numpy as np

mean = np.zeros((2,2,1))
xi = Input(shape=(2,2,1), name="input")
p = Lambda(lambda image, mu: image - mu, arguments={'mu':mean})(xi)
model = Model(inputs=[xi], outputs=[p])

print("Arguments to save : {}".format(model.layers[1].arguments))
model.save("titi.h5")
```

fails with "ValueError: can only convert an array of size 1 to a Python scalar"

The issue comes from models.py:get_json_type which invokes obj.item in case of a numpy object but this fails if the object is indeed a numpy array. 

This pull requests propose to handle specifically the case the serialized object is a ndarray by converting it into a list. Deserialization is done only in from_config of Lambda layers. This is a proposal, I do not master the API of Keras so guidelines for improving this PR are welcomed.

### CODE REVIEW

File: keras/layers/core.py [hunk 1]
```diff
@@ -720,6 +720,15 @@ def from_config(cls, config, custom_objects=None):
         else:
             output_shape = config['output_shape']
 
+        # If arguments were numpy array, they have been saved as
+        # list. We need to recover the ndarray
+        if 'arguments' in config:
+            for k in config['arguments']:
```
Comment (fchollet) [inline keras/layers/core.py]: Better to use a more explicit variable name, like `key`

File: keras/layers/core.py [hunk 2]
```diff
@@ -720,6 +720,15 @@ def from_config(cls, config, custom_objects=None):
...
+                if isinstance(config['arguments'][k], dict) and \
```
Comment (fchollet) [inline keras/layers/core.py]: Do not use backslashes to break lines.

File: keras/layers/core.py [hunk 3]
```diff
@@ -720,6 +720,13 @@ def from_config(cls, config, custom_objects=None):
         else:
             output_shape = config['output_shape']
 
+        # If arguments were numpy array, they have been saved as
+        # list. We need to recover the ndarray
+        if 'arguments' in config:
+            for key in config['arguments']:
+                if isinstance(config['arguments'][k], dict) and ('type' in config['arguments'][k]) and (config['arguments'][k]['type'] == 'ndarray'):
```
Comment (fchollet) [inline keras/layers/core.py]: Line way too long... please break it down (no backslashes, naturally).

**Comment by jeremyfix** [comment]
I would like to bring a more complete test script 



```
from keras.layers import Input, Lambda
from keras.models import Model, load_model
import numpy as np
from numpy.testing import assert_allclose

mean = np.random.random((2,2,1))
std = np.random.random((2,2,1))
xi = Input(shape=(2,2,1), name="input")
p = Lambda(lambda image, mu, std: (image - mu)/std, arguments={'mu':mean, 'std': std})(xi)
model = Model(inputs=[xi], outputs=[p])
model.save("titi.h5")

del model

model = load_model("titi.h5")

assert_allclose(mean, model.layers[1].arguments['mu'])
assert_allclose(std, model.layers[1].arguments['std'])
```

File: keras/layers/core.py [hunk 4]
```diff
@@ -720,6 +720,19 @@ def from_config(cls, config, custom_objects=None):
...
+        if 'arguments' in config:
+            for key in config['arguments']:
+                if isinstance(config['arguments'][key], dict):
+                    arg_dict = config['arguments'][key]
+                    if 'type' in arg_dict and arg_dict['type'] == 'ndarray':
+                        # Overwrite the argument with its numpy translation
+                        config['arguments'][key] = np.array(arg_dict['value'])
+                    else:
+                        warnings.warn('Unhandled dictionnary data in arguments {}'
+                                      .format(arg_dict))
```
Comment (fchollet) [inline keras/layers/core.py]: This warning does not seem necessary

**Comment by fchollet** [comment]
LGTM

**Pull request merged**